packages/contracts/docs/osx/02-how-to-guides/02-plugin-development/04-upgradeable-plugin/05-updating-versions.md

```diff
@@ -0,0 +1,357 @@
+  /// @dev The initialization of `SimpleStorageBuild1` has already happened.
+  function initializeFromBuild1(address _account) external reinitializer(2) {
+    account = _account;
+  }
+
+  function storeNumber(uint256 _number) external auth(STORE_PERMISSION_ID) {
+    number = _number;
+  }
+
+  function storeAccount(address _account) external auth(STORE_PERMISSION_ID) {
+    account = _account;
+  }
+}
+```
+
+### 2. Write a new Plugin Setup contract
+
+In order to do update a plugin, we need a `prepareUpdate()` function in our Plugin Setup contract which points the functionality to a new build, as we described in the ["How to create a subsequent build implementation to an Upgradeable Plugin" guide](./04-subsequent-builds.md). The `prepareUpdate()` function must transition the plugin from the old build state into the new one so that it ends up having the same permissions (and helpers) as if it had been freshly installed.
+
+In contrast to the original build 1, build 2 requires two input arguments: `uint256 _number` and `address _account` that we decode from the bytes-encoded input `_data`.
+
+```solidity
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity 0.8.17;
+
+import {PermissionLib} from '@aragon/osx/core/permission/PermissionLib.sol';
+import {PluginSetup, IPluginSetup} from '@aragon/osx/framework/plugin/setup/PluginSetup.sol';
+import {SimpleStorageBuild2} from './SimpleStorageBuild2.sol';
+
+/// @title SimpleStorageSetup build 2
+contract SimpleStorageBuild2Setup is PluginSetup {
+  address private immutable simpleStorageImplementation;
+
+  constructor() {
+    simpleStorageImplementation = address(new SimpleStorageBuild2());
+  }
+
+  /// @inheritdoc IPluginSetup
+  function prepareInstallation(
+    address _dao,
+    bytes memory _data
+  ) external returns (address plugin, PreparedSetupData memory preparedSetupData) {
+    (uint256 _number, address _account) = abi.decode(_data, (uint256, address));
+
+    plugin = createERC1967Proxy(
+      simpleStorageImplementation,
+      abi.encodeWithSelector(SimpleStorageBuild2.initializeBuild2.selector, _dao, _number, _account)
+    );
+
+    PermissionLib.MultiTargetPermission[]
+      memory permissions = new PermissionLib.MultiTargetPermission[](1);
+
+    permissions[0] = PermissionLib.MultiTargetPermission({
+      operation: PermissionLib.Operation.Grant,
+      where: plugin,
+      who: _dao,
+      condition: PermissionLib.NO_CONDITION,
+      permissionId: SimpleStorageBuild2(this.implementation()).STORE_PERMISSION_ID()
+    });
+
+    preparedSetupData.permissions = permissions;
+  }
+
+  /// @inheritdoc IPluginSetup
+  function prepareUpdate(
+    address _dao,
+    uint16 _currentBuild,
+    SetupPayload calldata _payload
+  )
+    external
+    pure
+    override
+    returns (bytes memory initData, PreparedSetupData memory preparedSetupData)
+  {
+    (_dao, preparedSetupData);
+
+    if (_currentBuild == 0) {
```

Comment:
Where there is no prior build, this value would be `0`. The very first buildId that can be upgraded from should be `1` as far as I can tell.

Reply:
Simple storage plugin update - added to notes!